src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/PostgreSqlEngine.java

```diff
@@ -832,4 +847,38 @@ protected ResultIterator createResultIterator(PreparedStatement ps) throws Datab
     protected QueryExceptionHandler getQueryExceptionHandler() {
         return PG_QUERY_EXCEPTION_HANDLER;
     }
+
+    /**
+     * Helper method that logs the time while the Postgres {@link Statement update query} is being executed.
+     * <br>
+     * This will help identify cases via logs where Postgres database is already locked due to an external query being
+     * executed and the current one is waiting for that one to finish first.
+     *
+     * @param function      The {@link Callable function} that will executes Postgres {@link Statement query}.
+     * @param queryLog      The query in {@link String} format which is being executed. Will be used in the logs.
+     * @throws SQLException If the {@link Statement query} fails during its execution.
+     * @implNote It prints a log every 5 minutes so the user is aware that the query is still being executed.
+     */
+    private<T> T executeUpdateQuery(final Callable<T> function, final String queryLog) throws SQLException {
+        final long startMilliSeconds = System.currentTimeMillis();
+        final Runnable printLog = () -> {
+            final long secondsPassed = (System.currentTimeMillis() - startMilliSeconds) / 1000;
+            this.logger.info("Database is still executing for '{}s' a query.\n" +
+                "Are you sure the Database is not blocked due to another query being executed?",
+                secondsPassed
+            );
+            this.logger.trace("The query still being executed for '{}s' is: '{}'.\n", secondsPassed, queryLog);
+        };
+        final ScheduledFuture<?> scheduledFutureLog = this.executor.scheduleAtFixedRate(printLog, 5L, 5L, TimeUnit.MINUTES);
```

Comment:
What happens when this is called concurrently or when a task is already scheduled to run?

Reply:
Where do you see it as synchronous? Even if it has any lock inside that makes it synchronous it is probably only synchronous on new updates done with the same statement. However, we do a `this.conn.createStatement()` that creates a new Statement object per connection. Am I missing something?

Reply:
Well the new method `com.feedzai.commons.sql.abstraction.engine.impl.PostgreSqlEngine#executeUpdateQuery` will only return when the `Statement` is executed so that makes it synchronous. WDYT?